src/js/tracker.js

```diff
@@ -1112,21 +1095,20 @@
 		function getOptimizelyXSummary() {
 			var state = getOptimizelyXData('state');
 			var experiment_ids = state.getActiveExperimentIds();
-			var experiments = getOptimizelyXData('data', 'experiments');
 			var visitor = getOptimizelyXData('visitor');
 
 			return map(experiment_ids, function(activeExperiment) {
 				var variation = state.getVariationMap()[activeExperiment];
-				var variationName = variation.name;
-				var variationId = variation.id;
-				var visitorId = visitor.visitorId;
+				var variationName = (variation && variation.name && variation.name.toString()) || null;
+				var variationId = variation && variation.id;
+				var visitorId = (visitor && visitor.visitorId && visitor.visitorId.toString()) || null;
 				return {
-					experimentId: parseInt(activeExperiment),
+					experimentId: parseInt(activeExperiment) || null,
 					variationName: variationName,
-					variation: parseInt(variationId),
+					variation: parseInt(variationId) || null,
 					visitorId: visitorId
```

Comment:
I suppose you could also check the type of visitorId.  It depends how paranoid you want to be.

Reply:
I'm toString'ing or nulling this just above. I think that should be enough to ensure we end up with a string.